src/Controller/OgAdminMembersController.php

```diff
@@ -21,13 +53,117 @@ class OgAdminMembersController extends ControllerBase {
     return Views::getView('og_members_overview')->executeDisplay('default', $arguments);
   }
 
+  /**
+   * Displays add membership links for available membership types.
+   *
+   * Returns default membership type if that's all that exists.
+   *
+   * @return array
+   *   A render array for a list of the node types that can be added; however,
+   *   if there is only one node type defined for the site, the function
+   *   will return the default add member form.
+   */
+  public function addPage(RouteMatchInterface $route_match) {
+    $membership_type_list = $this->entityTypeManager()->getStorage('og_membership_type')->loadMultiple();
+    if ($membership_type_list && count($membership_type_list) == 1) {
+      return $this->addForm($route_match);
+    }
+
+    $build = [
+      '#theme' => 'entity_add_list',
+      '#bundles' => [],
+    ];
+
+    $bundle_entity_type = $this->entityTypeManager()->getDefinition('og_membership_type');
+    $build['#cache']['tags'] = $bundle_entity_type->getListCacheTags();
+
+    $group = $this->getGroupFromRoute($route_match);
+
+    $add_link_params = [
+      'group' => $group->id(),
+      'entity_type_id' => $group->getEntityType()->id(),
+    ];
+
+    foreach ($membership_type_list as $membership_type_id => $og_membership_type) {
+      $add_link_params['membership_type'] = $membership_type_id;
+      $build['#bundles'][$membership_type_id] = [
+        'label' => $og_membership_type->label(),
+        'add_link' => Link::createFromRoute($og_membership_type->label(), "entity.og_membership.add_form", $add_link_params),
+      ];
+    }
+
+    return $build;
+  }
+
+  /**
+   * Provides the add member submission form.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match service.
+   *
+   * @return array
+   *   The member add form.
+   */
+  public function addForm(RouteMatchInterface $route_match) {
+    /** @var \Drupal\og\Entity\OgMembershipType $membership_type */
+    $membership_type = $route_match->getParameter('membership_type');
+    if (is_object($membership_type)) {
+      $membership_type = $membership_type->id();
+    }
+    if ($membership_type === NULL) {
+      $membership_type = 'default';
+    }
+
+    $group = $this->getGroupFromRoute($route_match);
+
+    /** @var \Drupal\og\Entity\OgMembership $og_membership */
+    $og_membership = OgMembership::create([
+      'type' => $membership_type,
+      'entity_type' => $group->getEntityType()->id(),
+      'entity_id' => $group->id(),
+    ]);
+
+    return $this->entityFormBuilder()->getForm($og_membership, 'add');
+  }
+
+  /**
+   * Determines the group associated with a route using this controller.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match service.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface
+   *   The group entity associated with the route.
+   */
+  protected function getGroupFromRoute(RouteMatchInterface $route_match) {
```

Comment:
@jludwig 
As I see, this member function is basically static, as `$this->entityTypeManager` could come statically.
And it's very general-purpose, what if we move it to:
```
/**
 * A static helper class for OG.
 */
class Og {
```

@amitaibu does it make sense? review the review please :), as a totally different approach might make sense, what I suggest is merely a little refactor.